content/docs/reference/design-docs/editoast-errors/_index.en.md

```diff
@@ -295,35 +295,62 @@ struct CoreError {
 
 Note: the `error` field is kept as a `serde_json::Value` and not parsed (even though its format is standard) as we're not supposed to perform any kind of analysis or recovery on it. If we end up parsing it in the future, that means we need a stronger mapping between Core errors and what editoast expects. The red flag will be more obvious if we end up manipulating a JSON dict instead of a proper structure.
 
-### Implementation plan
+### Why do we need a derive macro?
+
+The main issue with our error system is that the types we manipulate do not serialize to the error format we want. For example, an error defined like so:
+
+```rust
+#[derive(Debug, thiserror::Error)]
+#[error("{cause}")]
+struct MyError { cause: String, fix: String }
+```
+
+shouldn't be serialized as:
+
+```json
+{
+  "cause": "Emperor Zurg",
+  "fix": "Buzz Lightyear"
+}
+```
+
+like `serde::Serialize` would do, but as:
+
+```json
+{
+  "error_type": "editoast:MyError",
+  "status": 500,
+  "message": "Emperor Zurg",
+  "context": {
+    "cause": "Emperor Zurg",
+    "fix": "Buzz Lightyear",
+  }
+}
+```
+
+making `serde` basically useless for our errors. On top of that, since by design the derive macros of `utoipa` (`ToSchema`, `IntoResponse` especially) interpret the type structure like `serde` would do, we can't rely on them either. Therefore we need a custom derive-macro to convey the structural information of the type at runtime.
```

Comment:
I believe we still want `serde`, just not the automatic derive.

```suggestion
making `derive(serde::Serialize)` basically useless for our errors. ...
```

[Here are some ideas](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cb3f0255526cd98101bbb982112b9133) for not codegen the serialize implementation.

Reply:
[`1adee72` (#237)](https://github.com/OpenRailAssociation/osrd-website/pull/237/commits/1adee729633c161b3ed9d50b1512541e4e7c6cfb)

I agree it's probably a good approach to generate in the macro just what's necessary to get enough information about the type structure. We should indeed factorize the implementations of `IntoReponse` and `Serialize`/`axum::IntoResponse`.